### ISSUE TITLE
roc_auc should return NA with no cases or controls

### ISSUE THREAD

**Issue by topepo** [tidymodels/yardstick#132]
``` r
library(tidymodels)
#> Registered S3 method overwritten by 'xts':
#>   method     from
#>   as.zoo.xts zoo
#> ── Attaching packages ─────────────────────────────────────────────────────────── tidymodels 0.0.4 ──
#> ✓ broom     0.5.2          ✓ recipes   0.1.9     
#> ✓ dials     0.0.4          ✓ rsample   0.0.5     
#> ✓ dplyr     0.8.3          ✓ tibble    2.1.3     
#> ✓ ggplot2   3.2.1          ✓ tune      0.0.1     
#> ✓ infer     0.5.1          ✓ workflows 0.1.0     
#> ✓ parsnip   0.0.4.9000     ✓ yardstick 0.0.4     
#> ✓ purrr     0.3.3
#> ── Conflicts ────────────────────────────────────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard()    masks scales::discard()
#> x dplyr::filter()     masks stats::filter()
#> x dplyr::lag()        masks stats::lag()
#> x ggplot2::margin()   masks dials::margin()
#> x recipes::step()     masks stats::step()
#> x recipes::yj_trans() masks scales::yj_trans()
data(two_class_example)
two_class_example %>% 
  dplyr::filter(truth == "Class1") %>% 
  roc_auc(truth, Class1)
#> Error in roc.default(response, predictor, auc = TRUE, ...): No control observation.
```

<sup>Created on 2020-01-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

It would be better to return `NA` with a warning.

**Comment by DavisVaughan** [comment]
I think the main change to make is to `roc_auc_binary()`. It should have two checks:

- if `truth` doesn't have any `lvl[1]` values, then `NA_real_` should be returned with a warning that there were no events
- if `truth` doesn't have any of `lvl[2]` values, then `NA_real_` should be returned with a warning that there were no controls

At first glance it looks like this will be enough to fix macro/micro/macro_weighted multiclass cases too, since they use `roc_auc_binary()` internally

It looks like it will almost be enough to fix `roc_auc_hand_till()`. It uses `roc_auc_binary()`, but then the result is stored in `auc_val`, which is run through `if (auc_val < 0.5) {}`. If `auc_val = NA_real_` then this will cause an error, so you'd need to modify that function to exit early if `auc_val` is `NA`

Tests should be added for all binary / hand_till / macro / micro / macro_weighted variations

**Comment by DavisVaughan** [comment]
Closed by #138
